vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java

```diff
@@ -1,249 +1,124 @@
-        final Long size = 4_611_686_018_427_387_906L;
-
-        // Act
-        final String retval = Formats.toFixLengthSizeUnit(size);
-
-        // Assert result
-        Assert.assertEquals("4194304t", retval);
-    }
-
-    @Test
-    public void toMBInputNegativeOutputNotNull() {
-
-        // Arrange
-        final long bytes = -8L;
+    public void testShortName() {
+        Assert.assertEquals("!", shortName("!", 1, 6));
+        Assert.assertEquals("!...", shortName("!!!!!!!!!!", 4, 0));
 
-        // Act
-        final String retval = Formats.toMB(bytes);
-
-        // Assert result
-        Assert.assertEquals("NaN", retval);
+        thrown.expect(StringIndexOutOfBoundsException.class);
+        shortName("!!!!!!!!!!", 0, -19);
     }
 
     @Test
-    public void toMBInputPositiveOutputNotNull() {
-
-        // Arrange
-        final long bytes = 8L;
-
-        // Act
-        final String retval = Formats.toMB(bytes);
-
-        // Assert result
-        Assert.assertEquals("0m", retval);
+    public void testToFixLengthSizeUnit() {
+        Assert.assertEquals("NaN", toFixLengthSizeUnit(null));
+        Assert.assertEquals(" 123", toFixLengthSizeUnit(123L));
+        Assert.assertEquals(" 120k", toFixLengthSizeUnit(123_456L));
+        Assert.assertEquals(" 117m", toFixLengthSizeUnit(123_456_789L));
+        Assert.assertEquals(" 114g", toFixLengthSizeUnit(123_456_789_000L));
+        Assert.assertEquals("4194304t",
+          toFixLengthSizeUnit(4_611_686_018_427_387_906L));
     }
 
     @Test
-    public void toSizeUnitInputNegativeOutputNotNull() {
-
-        // Arrange
-        final Long size = -1023L;
-
-        // Act
-        final String retval = Formats.toSizeUnit(size);
-
-        // Assert result
-        Assert.assertEquals("-1023", retval);
+    public void testToMB() {
+        Assert.assertEquals("NaN", toMB(-8L));
+        Assert.assertEquals("0m", toMB(8L));
+        Assert.assertEquals("114g", toMB(123_456_789_000L));
     }
 
     @Test
-    public void toSizeUnitInputNullOutputNotNull() {
-
-        // Arrange
-        final Long size = null;
-
-        // Act
-        final String retval = Formats.toSizeUnit(size);
-
-        // Assert result
-        Assert.assertEquals("NaN", retval);
+    public void testToSizeUnit() {
+        Assert.assertEquals("-1023", toSizeUnit(-1023L));
+        Assert.assertEquals("NaN", toSizeUnit(null));
+        Assert.assertEquals("44k", toSizeUnit(45_312L));
+        Assert.assertEquals("117m", toSizeUnit(123_456_789L));
+        Assert.assertEquals("114g", toSizeUnit(123_456_789_000L));
+        Assert.assertEquals("112t", toSizeUnit(123_456_789_000_000L));
     }
 
     @Test
-    public void toSizeUnitInputPositiveOutputNotNull() {
-
-        // Arrange
-        final Long size = 45_312L;
+    public void testToSizeUnitWithColor() {
+        final LongWarning warning = new LongWarning();
+        Assert.assertEquals("123", toSizeUnitWithColor(123L, warning));
```

Comment:
Couldn't you just inline `new LongWarning()` here?

Reply:
That would take the line over 80 chars.

Reply:
Where has this 80 char limit come from? It's ridiculously short...

Reply:
From a previous PR comment and the style guide.

Reply:
You could still inline and then split the assert over two lines?